sdk/src/Services/S3/Custom/Transfer/Internal/SimpleUploadCommand.cs

```diff
@@ -41,12 +41,18 @@ internal partial class SimpleUploadCommand : BaseCommand
         IAmazonS3 _s3Client;
         TransferUtilityConfig _config;
         TransferUtilityUploadRequest _fileTransporterRequest;
+        long _totalTransferredBytes;
+        private readonly long _cachedContentLength;
 
         internal SimpleUploadCommand(IAmazonS3 s3Client, TransferUtilityConfig config, TransferUtilityUploadRequest fileTransporterRequest)
         {
             this._s3Client = s3Client;
             this._config = config;
             this._fileTransporterRequest = fileTransporterRequest;
+            
+            // Cache content length immediately while stream is accessible to avoid ObjectDisposedException in failure scenarios
+            this._cachedContentLength = this._fileTransporterRequest.ContentLength;
```

Comment:
we actually do this in multi part upload command too https://github.com/aws/aws-sdk-net/blob/9999d236f4d8babf2464918d5049ac9e816c3223/sdk/src/Services/S3/Custom/Transfer/Internal/MultipartUploadCommand.cs#L83